src/block/mod.rs

```diff
@@ -212,3 +217,26 @@ fn is_valid_block_file(path: &Path) -> bool {
         false
     }
 }
+
+fn extract_global_slot_since_genesis(file_path: &PathBuf) -> GlobalSlotNumberV1 {
+    let mut file = File::open(file_path)
+        .unwrap_or_else(|err| panic!("failed to open file: {}", err));
+    let mut contents = Vec::new();
+    file.read_to_end(&mut contents)
+        .unwrap_or_else(|err| panic!("failed to read file: {}", err));
+
+    let regex = Regex::new(r"global_slot_since_genesis=(\d+)")
+        .unwrap_or_else(|err| panic!("failed to create regex: {}", err));
+
+    let contents_str = String::from_utf8_lossy(&contents);
+
+    if let Some(captures) = regex.captures(&contents_str) {
+        if let Some(slot_str) = captures.get(1) {
+            if let Ok(slot) = slot_str.as_str().parse::<u32>() {
+                return GlobalSlotNumberV1::from(slot);
+            }
+        }
+    }
+
+    panic!("failed to extract global_slot_since_genesis from file");
+}
```

Comment:
We definitely should not read the entire file again just to get the slot number and we do not need to use a regex. That's what `serde_json::from_str::<BlockLog>(&str)` is for in `impl PrecomputedBlock`.